### DISCUSSION TITLE
FEATURE｜优化搜索Java时的UI行为

### DISCUSSION THREAD

**Discussion by Pastechn** [MCSLTeam/MCSL2#74] [功能建议] [answered]
现象描述：搜索 Java 的时候整个界面卡住（窗口无响应），然后等了很长一段时间才反应过来。

看了一下实现，虽然用上了 `QThread`，但是搜索线程开始运行后主线程貌似是通过 `sleep()` 来等待子线程完成搜索。

可以试试在开始搜索时再显示一个正在搜索的窗口（或控件），让子线程通过回调的方式通知主线程搜索完成并关闭刚才的窗口（控件）。

~~纯小白，仅供参考，其实不仅搜索 Java 的时候会卡住，选择下载源的时候也会这样~~

**Comment by AresConnor** [comment]
确实可以，可以试试在QThread里创建一个pyqtSignal，在主线程订阅(connect)这个信号，当子线程QThread运行结束时在通过pyqtSigna;l.emit方法来发出事件

**Comment by lxhtt** [comment]
有空就改，已收到

**Comment by lxhtt** [comment, marked as answer]
本问题已经在最新构建版修复。
